### DISCUSSION TITLE
Do I create the git storage repo or does wiki.js do it?  If I do it, where should it be initialized?

### DISCUSSION THREAD

**Discussion by Vaslo** [requarks/wiki#5458] [Help / Questions] [answered]
I have the unRaid container and I've been trying to get the Git/Github storage working.  I'm not clear about the workflow with creating the repo.  I've tried to follow the directions on the wiki site for Git but I just can't seem to get it working - maybe those are for local installs.  My container looks like this:
![image](https://user-images.githubusercontent.com/7269272/178089030-eac19cff-2014-46db-90f5-63ac47505488.png)
The `data` folder is the only persistent item.  I originally created the SSH key in there, but now that I've read through the comments I realize I shouldn't make the key in the container.
These are the three folders in that persistent:
![image](https://user-images.githubusercontent.com/7269272/178089169-e3f74fac-ec43-4f3f-b39d-47abdf07c50a.png)
My question is:  Do I create the git repo somewhere?  If so where is the best location and which folders should be included?  Or does Wiki.js initialize this repo somehow?  How does that work and where does it create it?

Sorry for the beginner question - otherwise I love the app, thanks for the hard work.

**Comment by jarrodlovas** [comment, marked as answer]
You need to link it to a git repo using the GUI in https://domain/a/storage

**Reply by Vaslo** [reply]
But who creates the repo?  Do I do it? Does the app?  What do you mean with the link you posted?  With the directories I posted, is the repo created at the top of the data folder, or somewhere else.  I'm used to creating repos at the top of a set of folders I want to add but do I do it elsewhere and just add manually?

**Reply by jarrodlovas** [reply]
You need to create the repo.
The link was just telling you where in the admin portal you can view storage methods. 
For example, if your domain is a domain.com you would go to https://wiki.yourdomain.com/a/storage

Here is where my git folder was created when I switched to git storage
![image](https://user-images.githubusercontent.com/47882472/178106371-355ab93e-b838-45fb-ad1a-d48972c808fb.png)

**Reply by jarrodlovas** [reply]
That folder was automatically created, I didn't need to clone the repo in that folder or anything.

**Reply by Vaslo** [reply]
This is helpful.    One more question - since my data folder is the only persistent folder in my unRaid, I would think I'd need to make my repo there and point to it.  Is that accurate?  Or should this app be managing it more for me?  Thanks for your help and patience.

**Reply by jarrodlovas** [reply]
How did you do your installation? Via docker?

**Reply by fnafox** [reply]
I believe Vaslo misunderstood it as running git init locally which is not necessary at all. If the intention is to dump wiki content to disk then just use local file system as a storage option. Git storage functions by syncing it to a remote git repository.

You need to create the repo in the git software you're using, be that remote like github or some locally hosted git instance.
Once the git repo is created, connect to it using wikijs by providing the necessary information in the git storage interface.
There's nothing you need to do or setup in the command line for this to work, nor does it require some special setup of your local storage.
